py/serial_keel.py

```diff
@@ -197,18 +197,25 @@ async def _read(self) -> None:
 
         while True:
             response = await self.skws.read()
+
+            # When we get messages, they may come utf-8 encoded. Those are not very nice
+            # to read. Try to decode them and replace the original stream
+            try:
+                message_ref = response.get('Ok', {}).get('Async', {}).get('Message', {})
+                decoded_message = bytes(message_ref['message']).decode('utf-8')
+                message_ref['message'] = decoded_message.strip()
+            except (TypeError, KeyError):
+                # Just continue and append response as is
+                pass
+
             self.logger.debug(f'Response: {response}')
 
             if 'Ok' in response:
                 value = response['Ok']
 
                 if 'Async' in value:
-                    message = value['Async']['Message']
-                    endpoint = message['endpoint']['id']
-
-                    # TODO: We should expose to the user whether they want bytes only or
-                    # if they want to decode.
-                    message = bytes(message['message']).decode('utf-8')
+                    message = value['Async']['Message']['message']
```

Comment:
The mocks as usual came to my rescue.  Anyways, I'm not sure if this was your intention based on the TODO here.

Reply:
Great to hear the mocks bring value.
The TODO relates to allowing users to make a choice whether they want to opt-in/opt-out to decoding incoming messages. I was thinking about use cases for SK for binary data instead of strings.
Added #6 to remember it.